duckduckgo/BrowserServicesKit
# Support internal config feature state

Please review the release process for BrowserServicesKit [here](https://app.asana.com/0/1200194497630846/1200837094583426).

**Required**:

Task/Issue URL: https://app.asana.com/0/72649045549333/1203676877763836
iOS PR:  https://github.com/duckduckgo/iOS/pull/1565
macOS PR: https://github.com/duckduckgo/macos-browser/pull/1054
What kind of version bump will this require?: Minor

**Optional**:

Tech Design URL: https://app.asana.com/0/481882893211075/1203938569856702
CC:

**Description**:

On the back of the changes made in #270 to introduce a FeatureFlagger + autofill Privacy-Config subfeatures to BSK, this PR adds support for the newly added `internal` case of the `state` enum field of both the top level `features` and their nested `features`.

This change will also allow us to release changes that will only be visible to internal users, then enabled them for end users remotely.



**Steps to test this PR**:
1. Check out the iOS & macOS PRs above
2. Check that any features configured using the privacy-config still work as expected (particularly autofill for iOS).
3. Try changing the FeatureFlag configurations from iOS to make sure that they give the appropriate level of visibility.



**OS Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
